cockatrice/src/deck_loader.cpp

```diff
@@ -288,13 +289,25 @@ QString DeckLoader::getCardZoneFromName(QString cardName, QString currentZoneNam
     return currentZoneName;
 }
 
-QString DeckLoader::getCompleteCardName(const QString cardName) const
+QString DeckLoader::guessCompleteCardName(const QString cardName) const
 {
+    static const QRegularExpression evenSimplerExp("(\\s+|\\/\\/.*)"); // matches whitespace or double names
+
     if (db) {
-        CardInfoPtr temp = db->getCardBySimpleName(cardName);
+        const CardInfoPtr temp = db->getCardBySimpleName(cardName);
         if (temp) {
             return temp->getName();
         }
+
+        // could not find the card, so simplify more
+        QString simpleCardName = CardInfo::simplifyName(QString(cardName).remove(evenSimplerExp));
+        for (const CardInfoPtr &card : db->getCardList()) {
```

Comment:
This iteration looks like it's super slow, as it's iterating 50k cards and then applying a regular expression to each... is this the best approach?

Reply:
it's actually applying around 4 regexes, and it's only actually used if a card is not found in the database. I considered storing the table but it's important to update the table when it's reloaded but it's also not used often enough to warrant always creating this new table on a reload.

Reply:
I just tested it, it's quite slow actually, it takes like 2 seconds for only 12 cards

Reply:
that is 12 unknown cards, cards that are immediately recognized are snappy as normal